Added "reset" button - Fixes #40

drwm-base opened the pull request:
Hey @pacdiv 

As of now I have implemented a "Reset Conditions" button which sets the state back to it's instantiation settings, as well as clears all conditions from the core using deleteCondition(). I have also added an onResetButtonClick() function to handle the resetting process, and I have used a while loop to remove all of the conditions in the core using deleteCondition(), due to the fact that deleteCondition() reduces the size of the array every time it deletes. 

I have also updated and passed all tests and code formatting checks (prettier). Please let me know if there is anything else you need me to change, add, or remove!

NOTE: One interesting thing, for some reason git wouldn't let me commit without making a lot of changes to formatting in a lot of your files. I believe it has something to do with prettier in your package.json file, under the pre-commit part. It made the pull request 53 files changed instead of 2 or 3. I don't really know why, but it mostly changed single quotes to double quotes in your files and wouldn't let me commit otherwise.

drwm-base committed:
added reset button

drwm-base committed:
added reset functionality

also updated jest snapshots

drwm-base committed:
made reset button transparent

drwm-base committed:
conformed to prettier formatting

drwm-base committed:
fixed reset functionality for multiple conditions

drwm-base committed:
Update Playground.js

drwm-base commented:
My real changes are in playground.js only

pacdiv commented:
Thanks for submitting your PR! 🤗 

No worries about these prettier modifications. My mistake, I should have seen this earlier.
I'll take a look at everything and get back to you in a few minutes.

pacdiv commented on src/components/playground/Playground/Playground.js:
```diff
@@ -110,6 +110,22 @@ class Playground extends Component {
+    })
+    while (this.core.getChunks().length !== 0) {
+      this.core.deleteCondition(0)
+    }
```
This is nice @drwm-base! 👍 
It removes all the conditions as expected!

However, it resets the flags too (line 118). Am I right? Is this may delete an enabled _“global flag”_?

Also, have a noticed that the core manages the adds, updates and single deletions? Maybe I should write this in guidelines, my bad. So, in your new code, the loop is in _Playground.js_, which makes the component _Playground_ responsible of its own _this.core_'s conditions array. What do you think about moving this responsibility to the _core.js_ file by creating in it a new method (_removeAllConditions_ for instance) and calling this new core instance method from _Playground.onResetConditionsButtonClick_?

drwm-base replied:
I thought that would be a good idea when I was writing, but I wasn't sure as to the scope that you wanted me to cover (ie just add a new button / changes to the core or just playground.js). I can move the functionality over to the core.js file, it would make things look cleaner as well. I will do that asap!

Thanks for the feedback!

drwm-base committed:
moved deleteAllConditions to core

- removed unnecessary props from setState function

- simplified deleteAllConditions to clear both this.chunks and this.regexChunks using length = 0 method. This is faster than deleting one at a time.

drwm-base committed:
prettier fixes

drwm-base commented:
Hey @pacdiv ,

I have updated the functions as you requested. Let me know if there is anything else you need me to change / add! 

I changed deleteAllConditions to just clear the arrays using length = 0. This is faster than looping through as it skips the mapping and filtering that are not required for deleting the whole array, and it still works!

pacdiv commented:
You're right! The looping wasn't necessary. I like what you've done! 🙂 

Actually, even using `.length = 0` might not be necessary by doing this:
```
this.chunks = []
this.capturedChunks = []
this.regexChunks = []
```

This would reset these arrays without using their `length` attribute.
Can you do this update?

After this, I think this feature will be good for a merge! 🚀 
Thank you so much for your updates so far, I really appreciate! 🙏

drwm-base committed:
fixed deleteAllConditions

drwm-base commented:
@pacdiv updated to your specs. Thanks again! If you would like more people to get involved in your project, I have a few friends looking for projects to get involved in for Hacktoberfest.

pacdiv commented:
About the reference issue, it's an interesting think you've pointed out. I try to avoid references by using Array.from, Array.filter, Array.map and Array.reduce. So, here, we can use `= []`.
But yes, if I use references some day, I'll consider the `.length = 0`.

Thank you so much for your help, I really appreciated! 🙏
